sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -647,6 +647,10 @@ object ColumnPruning extends Rule[LogicalPlan] {
     // Can't prune the columns on LeafNode
     case p @ Project(_, _: LeafNode) => p
 
+    case p @ NestedColumnAliasing(nestedFieldToAlias, attrToAliases)
+        if SQLConf.get.nestedSchemaPruningEnabled =>
```

Comment:
We need this check? It seems we've already checked this flag in https://github.com/apache/spark/pull/23964/files#diff-43334bab9616cc53e8797b9afa9fc7aaR36?

Reply:
Thanks. Yes. Since it's moved to there recently, we can remove this here.